File: files/CH-063.R
library(tidyverse)
library(readxl)
library(rebus)

input = read_excel("files/CH-063 Custom splitter.xlsx", range = "B2:B15")
test  = read_excel("files/CH-063 Custom splitter.xlsx", range = "D2:F15")

pattern = capture(digit(4) %R% "/" %R% digit(1,2) %R% "/" %R% digit(1,2)) %R%
  capture(alpha(1,2)) %R% capture(digit(1,2))


result = input %>%
  extract(col = "Info", into = c("Date", "Product", "Quantity"), pattern) %>%
  mutate(Quantity = as.numeric(Quantity))

identical(result, test)
# [1] TRUE

File: files/000-099/063/CH-063.py
import pandas as pd

input = pd.read_excel("CH-063 Custom splitter.xlsx", usecols="B", skiprows=1)
test = pd.read_excel("CH-063 Custom splitter.xlsx", usecols="D:F", skiprows=1)

pattern = r"(\d{4}/\d{1,2}/\d{1,2})([A-Za-z]{1,2})(\d{1,2})"

result = input["Info"].str.extract(pattern)
result.columns = ["Date", "Product", "Quantity"]
result["Quantity"] = pd.to_numeric(result["Quantity"])

print(result.equals(test)) # True